src/multiplex/client.rs

```diff
@@ -34,14 +34,80 @@ pub trait TagStore<Request, Response> {
     fn finish_tag(self: Pin<&mut Self>, r: &Response) -> Self::Tag;
 }
 
+pub trait PendingStore<T, Request>
+where
+    T: TryStream + Sink<Request> + TagStore<Request, T::Ok>,
+{
+    fn insert(self: Pin<&mut Self>, pending: Pending<T::Tag, T::Ok>);
+    fn take(
```

Comment:
Yeah, names I am happy to bikeshed. I don't like sent/complete since this is actually storing something or expecting you to store something. Let me revisit tomorrow.

Reply:
I think that expectation is made clear through the return value of the second function + in documentation.

Reply:
I follow what you're saying so I think maybe `sent`/`received` fits well?